File: parantier-api/src/main/java/com/mapo/palantier/topic/domain/Topic.java
package com.mapo.palantier.topic.domain;

import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Topic {

    private Long id;
    private String title;
    private String content;
    private String contentPlain;
    private Long authorId;
    private String authorName;
    private Integer viewCount;
    private Integer commentCount;
    private Boolean isPinned;
    private Boolean isActive;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: parantier-front/src/entities/topic/types/topic.ts
export interface Topic {
  id: number
  title: string
  content?: string
  contentPlain?: string
  authorId: number
  authorName: string
  viewCount: number
  commentCount: number
  isPinned: boolean
  isActive: boolean
  createdAt: string
  updatedAt: string
}

export interface TopicComment {
  id: number
  topicId: number
  authorId: number
  authorName: string
  content: string
  createdAt: string
  updatedAt: string
}
